src/subclass/widget.rs

```diff
@@ -214,6 +311,253 @@ impl<T: WidgetImpl + ObjectImpl> WidgetImplExt for T {
     //     }
     // }
 
+    fn parent_child_notify(&self, widget: &Widget, child_property: &glib::ParamSpec) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .child_notify
+                .expect("No parent class impl for \"child_notify\"");
+            let pspec_glib = glib::translate::mut_override(child_property.to_glib_none().0);
+            f(widget.to_glib_none().0, pspec_glib)
+        }
+    }
+
+    fn parent_composited_changed(&self, widget: &Widget) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .composited_changed
+                .expect("No parent class impl for \"composited_changed\"");
+            f(widget.to_glib_none().0)
+        }
+    }
+
+    fn parent_compute_expand(&self, widget: &Widget, hexpand_p: &mut bool, vexpand_p: &mut bool) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .compute_expand
+                .expect("No parent class impl for \"compute_expand\"");
+            f(
+                widget.to_glib_none().0,
+                *hexpand_p as i32 as *mut i32,
+                *vexpand_p as i32 as *mut i32,
```

Comment:
I think this is wrong. You're casting an integer to a pointer.

I'd do something like
```rust
let mut h: i32 = hexpand_p.to_glib();
f(&mut h);
*hexpand_p = from_glib(h);
```